vm/devices/firmware/firmware_uefi/src/service/nvram/mod.rs

```diff
@@ -348,20 +348,39 @@ impl NvramServices {
             use uefi_specs::uefi::nvram::vars::SECURE_BOOT;
 
             let (vendor, name) = SECURE_BOOT();
+            let attr: u32 = (EfiVariableAttributes::DEFAULT_ATTRIBUTES_VOLATILE).into();
+
+            let mut do_write = true;
+
             // Older versions of OpenHCL (and Hyper-V, closed-source HCL, etc. ) may have created
             // SecureBoot variable with NV attribute, which doesn't match the UEFI spec.  If this
-            // variable already exists, use its existing attributes.
-            let attr = match self.services.get_variable_ucs2(vendor, name).await {
-                Ok((existing_attr, _)) => existing_attr,
-                Err(_) => (EfiVariableAttributes::DEFAULT_ATTRIBUTES_VOLATILE).into(),
+            // variable already exists with the wrong attributes, delete it.
+            match self.services.get_variable_ucs2(vendor, name).await {
```

Comment:
Prefer expression-oriented variable assignment. This can be `let do_write = match`, and then the various match arms can return `true` or `false` as needed.

Reply:
> can return
can evaluate to

Reply:
on a more meta note though: maybe there's no need to do this "do_write" dance at all?

Why not unconditionally delete whatever existing var is there, and overwrite it with a fresh way, which is using the correct attributes?

Reply:
I thought it might be worth saving a round-trip to the NVRAM store, but I'm open to simplifying the code.

Reply:
If this is volatile, why is it in the store at all?

Reply:
Maybe two writes beats a read + write

Reply:
> If this is volatile, why is it in the store at all?

I am thinking about this

Reply:
> > If this is volatile, why is it in the store at all?
> 
> I am thinking about this

Maybe it's time to address this a better way; UEFI would need to be told via config flag that secure boot is controlled by config flags, with another flag to enable/disable secure boot, which could set a PCD.  In this mode, UEFI wouldn't consult the backing store to return the SecureBoot mode value.  Hyper-V, etc. would take a lot more time to fix up...

I think we need to land one version of this PR for now (to unblock developers).

Reply:
Note that there are a few other variables in this same category (DBDEFAULT and the VSM variables).  DbDefault is not a boolean